golem/task/tasksession.py

```
@@ -996,19 +985,24 @@ def check_provider_for_subtask(self, subtask_id) -> bool:
             return False
         return True
 
-    def check_requestor_for_task(self, task_id, additional_msg="") -> bool:
+    def check_requestor_for_task(self, task_id, additional_msg="") \
+            -> Optional[bool]:
         node_id = self.task_manager.comp_task_keeper.get_node_for_task_id(
             task_id)
+        if node_id is None:
+            return None
         if node_id != self.key_id:
             logger.warning('Received message about task %r from diferrent '
                            'node %r than expected %r. %s', task_id,
                            self.key_id, node_id, additional_msg)
             return False
         return True
 
-    def check_requestor_for_subtask(self, subtask_id) -> bool:
+    def check_requestor_for_subtask(self, subtask_id) -> Optional[bool]:

```

Would be nice to rewrite this to raise exceptions instead of this quirky triple return, but prolly outside of this PR.

---

In my opinion, exceptions should be raised in unexpected situations. This is totally expected. An enum would be better.